html/modules/custom/reliefweb_job_tagger/reliefweb_job_tagger.module

```diff
@@ -0,0 +1,261 @@
+    $form[$field]['#states'] = [
+      'visible' => [
+        'select[name="field_source"]' => $conditions,
+      ],
+      'required' => [
+        'select[name="field_source"]' => $conditions,
+      ],
+    ];
+  }
+
+  $form['ai_message']['#states'] = [
+    'invisible' => [
+      'select[name="field_source"]' => $conditions,
+    ],
+  ];
+}
+
+/**
+ * Get a list of trusted sources for a given account.
+ */
+function reliefweb_job_tagger_get_trusted_organizations($account = NULL) {
+  $sources = UserPostingRightsHelper::getUserPostingRights($account);
+  $sources = array_filter($sources, function ($info) {
+    if (isset($info['job']) && $info['job'] == 3) {
+      return TRUE;
+    }
+  });
+  $sources = array_keys($sources);
+
+  return $sources;
+}
+
+/**
+ * Implements hook_module_implements_alter().
+ */
+function reliefweb_job_tagger_module_implements_alter(&$implementations, $hook) {
+  switch ($hook) {
+    // Move our hook_entity_type_alter() implementation to the end of the list.
+    case 'node_presave':
+    case 'node_update':
+    case 'node_insert':
+      $group = $implementations['reliefweb_job_tagger'];
+      unset($implementations['reliefweb_job_tagger']);
+      $implementations['reliefweb_job_tagger'] = $group;
+      break;
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ */
+function reliefweb_job_tagger_node_presave(EntityInterface $node) {
+  // We don't want to react to configuration imports.
+  if (\Drupal::isConfigSyncing()) {
+    return;
+  }
+
+  if (!$node instanceof Node) {
+    return;
+  }
+
+  if ($node->bundle() != 'job') {
+    return;
+  }
+
+  if ($node->moderation_status->value != 'pending') {
+    $node->set('reliefweb_job_tagger_status', '');
+    return;
+  }
+
+  // Check permissions.
+  $user = \Drupal::currentUser();
+  if ($user->hasPermission('bypass ocha ai job tag')) {
+    return;
+  }
+
+  if (!$user->hasPermission('enforce ocha ai job tag')) {
+    return;
+  }
+
+  // Only queue it when fields are empty.
+  if (!$node->field_job_experience->isEmpty()) {
+    return;
+  }
+
+  if (!$node->field_career_categories->isEmpty()) {
+    return;
+  }
+
+  if (!$node->field_theme->isEmpty()) {
+    return;
+  }
+  if ($node->reliefweb_job_tagger_status->isEmpty()) {
+    $node->setNewRevision(FALSE);
+    $node->set('reliefweb_job_tagger_status', 'queue');
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_insert().
+ */
+function reliefweb_job_tagger_node_insert(Node $node) {
+  // We don't want to react to configuration imports.
+  if (\Drupal::isConfigSyncing()) {
+    return;
+  }
+
+  if ($node->hasField('reliefweb_job_tagger_status') && $node->get('reliefweb_job_tagger_status')->value == 'queue') {
+    reliefweb_job_tagger_queue_job($node);
+
+    $node->set('reliefweb_job_tagger_status', 'queued');
+    $node->setRevisionLogMessage('Job has been queued for tagging');
+    $node->setNewRevision(FALSE);
+    $node->updateLoadedRevisionId();
+    $node->save();
```

Comment:
Normally an entity should not be saved again in its own insert/update/delete hooks because it's still inside the DB transaction for the orignal `save()` call.

RW introduces an `after_save` hook in `Drupal\reliefweb_entities\BundleNodeStorage` that can be used for that. See `reliefweb_api_entity_after_save()` for an implementation example.

Otherwise, probably better to move the logic to the `reliefweb_job_tagger_node_presave` and remove the `$node->save()` altogether.

Same comment for the `reliefweb_job_tagger_node_update` below.

Reply:
Switched to reliefweb_api_entity_after_save